src/DeferredRender.js

```diff
@@ -2,31 +2,40 @@
 // @flow
 import * as React from 'react';
 
-let rIC;
-
-if (window.requestIdleCallback) {
-  rIC = window.requestIdleCallback;
-} else {
-  rIC = (handler, { timeout }) => setTimeout(handler, timeout);
+function createRIC() {
+  let rIC;
+
+  if (window.requestIdleCallback) {
+    rIC = (handler, options) => {
+      const id = requestIdleCallback(handler, options);
+      return () => cancelIdleCallback(id);
+    };
+  } else {
+    rIC = (handler) => {
+      const id = requestAnimationFrame(() => requestAnimationFrame(handler));
+      return () => cancelAnimationFrame(id);
```

Comment:
There are actually two ids, one for each for each of the rAF calls. When the first callback is updated, you need to store the second rAF's id and ensure that's canceled.